src/shared/hooks/useCardDetails.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState } from 'react'
+import getDetails from '../utils/getDetails/getDetails'
+import CardData from '../../models/CardData'
+import UseQueryParams from './useQueryParams'
+
+export default function UseCardDetails() {
+  const { id, clearParams } = UseQueryParams('details')
+  const [isLoading, setIsLoading] = useState<boolean>(true)
+  const [details, setDetails] = useState<CardData>()
+
+  const getData = async (id: string) => {
+    try {
+      setIsLoading(true)
+      const data = await getDetails(id)
+      setDetails(data)
+    } catch {
+      setDetails(undefined)
```

Comment:
when we set value it's better to use null, undefined  is used for unassigned a value